pkg/etcd/etcd.go

```diff
@@ -1116,6 +1158,50 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, promoted bool) (*v1.Node, error) {
+	var newCondition v1.NodeCondition
+	if promoted {
+		newCondition = v1.NodeCondition{
+			Type:               "EtcdIsVoter",
+			Status:             "True",
+			LastHeartbeatTime:  node.CreationTimestamp,
+			LastTransitionTime: node.CreationTimestamp,
+			Reason:             "EtcdIsAVoter",
+			Message:            "Etcd member is a voter",
+		}
+	} else {
+		newCondition = v1.NodeCondition{
+			Type:               "EtcdIsVoter",
+			Status:             "False",
+			LastHeartbeatTime:  node.CreationTimestamp,
+			LastTransitionTime: node.CreationTimestamp,
+			Reason:             "EtcdIsNotAVoter",
+			Message:            "Etcd member is not a voter",
+		}
+	}
+
+	for key, condition := range node.Status.Conditions {
+		if key == (len(node.Status.Conditions)-1) && condition.Type != "EtcdIsVoter" {
```

Comment:
Convention is to use `i` when ranging over a list or slice; `key` suggests that you're iterating over a map with  key:value pairs.

```suggestion
	for i, condition := range node.Status.Conditions {
		if i == (len(node.Status.Conditions)-1) && condition.Type != "EtcdIsVoter" {
```

Also, add a string constant for the `EtcdIsVoter` type.